src/pl/michaeli/service/ConfirmationResource.java

```diff
@@ -0,0 +1,220 @@
+
+		context.close();
+
+        GenericEntity<Reservation> myEntity = new GenericEntity<Reservation>(reservation) {};
+        return Response.status(200).entity(myEntity).build();
+    }
+		
+	private Boolean checkName(String name) {
+		Boolean isCorrect = Character.isUpperCase(name.charAt(0));
+        for(int i=1;i<name.length();i++) {
+            if(isCorrect) {
+                isCorrect = Character.isLowerCase(name.charAt(i));
+            }
+        }
+		return isCorrect;
+	}
+	
+	private Boolean checkSurname(String surname) {
+		if(surname.contains("-")) {
+			int index = surname.indexOf("-");
+			String firstPart=surname.substring(0, index);
+			String secondPart=surname.substring(index+1);
+			return (checkName(firstPart)&&checkName(secondPart));
+			
+		}else {
+			return checkName(surname);
+		}
+	}
+	
+	private double countTotalAmount(List<BookedSeat> bookedSeats, Date screeningDate) {
+		AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(AppConfig.class);
+		TicketTypesDAO ticketTypesDAO=context.getBean(TicketTypesDAO.class);
+		context.close();
+		double totalAmount=0;
+		for(int i=0; i<bookedSeats.size(); i++) {
+			totalAmount+=ticketTypesDAO.getPrice(bookedSeats.get(i).getTicketTypeId());
+		}
+		if(checkIsWeekend(screeningDate)) {
+			totalAmount = totalAmount + 4*bookedSeats.size();
+		}
+		return totalAmount;
+	}
+	
+	private Boolean checkIsWeekend(java.util.Date date) {
+		Boolean isWeekend = false;
+		Calendar cal = Calendar.getInstance();
+	    cal.setTime(date);
+	    switch(cal.get(Calendar.DAY_OF_WEEK)){
+	    	case Calendar.FRIDAY:
+	    		if(cal.get(Calendar.HOUR_OF_DAY)>=14)
+	                isWeekend=true;
+	            break;
+	        case Calendar.SATURDAY:
+	            isWeekend=true;
+	            break;
+	        case Calendar.SUNDAY:
+	            if(cal.get(Calendar.HOUR_OF_DAY)<23)
+	                isWeekend=true;
+	            break;
+	    }
+		return isWeekend;
+	}
+	
+	private Date countExpirationDate(Date screeningDate) {
+		Calendar cal = Calendar.getInstance();
+		cal.setTime(screeningDate);
+		cal.add(Calendar.MINUTE, -15);
+		Date expirationTime = cal.getTime();
+		return expirationTime;
+	}
+	
+	private Boolean checkIsAvailable(List<BookedSeat> seatsToBook,int screeningId) {
+		Boolean isAvailable=true;
+		AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(AppConfig.class);
+		BookedSeatsDAO bookedSeatsDAO=context.getBean(BookedSeatsDAO.class);
+		for(int i=0;i<seatsToBook.size();i++) {
+			BookedSeat seatToBook=seatsToBook.get(i);
+			if(bookedSeatsDAO.doesExist(seatToBook.getSeatId(), screeningId)) {
+				isAvailable=false;
+				break;
+			}
+		}
+		context.close();
+		return isAvailable;
+	}
+	
+	private Boolean checkIfFreeSpaceLeft(List<BookedSeat> seatsToBook,int screeningId) {
+		Boolean isFreeSpaceLeft=false;
+		int[][]placesInRoom = new int[5][10];
```

Comment:
Co to znaczy 5 albo 10? Skąd się wzięły te liczby?

Reply:
Trochę poszłam na łatwiznę z tymi miejscami... Wprowadzenie stałych trochę pomoże? :-)
static final int NUMBER_OF_ROWS_IN_SCREENING_ROOM = 5;
static final int NUMBER_OF_SEATS_IN_ROW_IN_SCREENING_ROOM = 10;

Reply:
Dobra stała ma dwie zalety - jest stała i jest dobra... No i jeszcze deskryptywna ;)